apps/src/blockly/customBlocks/googleBlockly/proceduresBlocks.js

```diff
@@ -0,0 +1,142 @@
+  {
+    type: 'procedures_defnoreturn',
+    message0: '%1 %2 %3 %4',
+    message1: '%1',
+    args0: [
+      {
+        type: 'field_label',
+        text: ' ',
+      },
+      {
+        type: useModalFunctionEditor ? 'field_label' : 'field_input',
+        name: 'NAME',
+        text: '',
+        spellcheck: false,
+      },
+      {
+        type: 'field_label',
+        name: 'PARAMS',
+        text: '',
+      },
+      {
+        type: 'input_dummy',
+        name: 'TOP',
+      },
+    ],
+    args1: [
+      {
+        type: 'input_statement',
+        name: 'STACK',
+      },
+    ],
+    style: 'procedure_blocks',
+    helpUrl: '%{BKY_PROCEDURES_DEFNORETURN_HELPURL}',
+    tooltip: '%{BKY_PROCEDURES_DEFNORETURN_TOOLTIP}',
+    extensions: [
+      'procedure_def_get_def_mixin',
+      'procedure_def_var_mixin',
+      'procedure_def_update_shape_mixin',
+      'procedure_def_context_menu_mixin',
+      'procedure_def_onchange_mixin',
+      'procedure_def_validator_helper',
+      'procedure_defnoreturn_get_caller_block_mixin',
+      'procedure_defnoreturn_set_comment_helper',
+      'procedure_def_set_no_return_helper',
+      'procedures_block_frame',
+    ],
+    mutator: 'procedure_def_mutator',
+  },
+  {
+    type: 'procedures_callnoreturn',
+    message0: '%1 %2',
+    args0: [
+      {type: 'field_label', name: 'NAME', text: '%{BKY_UNNAMED_KEY}'},
+      {
+        type: 'input_dummy',
+        name: 'TOPROW',
+      },
+    ],
+    nextStatement: null,
+    previousStatement: null,
+    style: 'procedure_blocks',
+    helpUrl: '%{BKY_PROCEDURES_CALLNORETURN_HELPURL}',
+    extensions: [
+      'procedures_edit_button',
+      'procedure_caller_get_def_mixin',
+      'procedure_caller_var_mixin',
+      'procedure_caller_update_shape_mixin',
+      'procedure_caller_context_menu_mixin',
+      'procedure_caller_onchange_mixin',
+      'procedure_callernoreturn_get_def_block_mixin',
+    ],
+    mutator: 'procedure_caller_mutator',
+  },
+]);
+
+// Respond to the click off a call block's edit button
+export const editButtonHandler = function () {
+  console.log('edit button clicked!');
+
+  // Eventually, this will be where we create a modal function editor.
+  // For now, just find the function definition block and select it.
+  const workspace = this.getSourceBlock().workspace;
+  const name = this.getSourceBlock().getFieldValue('NAME');
+  const definition = GoogleBlockly.Procedures.getDefinition(name, workspace);
+  if (definition) {
+    workspace.centerOnBlock(definition.id);
+    definition.select();
+  }
+};
+
+// This extension adds an edit button to the end of a procedure call block.
+const editExtension = function () {
+  if (useModalFunctionEditor) {
+    const button = new Blockly.FieldButton({
+      value: 'edit',
+      onClick: editButtonHandler,
+      colorOverrides: {button: 'blue', text: 'white'},
+    });
+    this.inputList[this.inputList.length - 1].appendField(button, 'EDIT');
```

Comment:
As things are written, no. The `procedures_callnoreturn` block we define above will always have a dummy input to contain the field that lists the name:

```
args0: [
      {type: 'field_label', name: 'NAME', text: '%{BKY_UNNAMED_KEY}'},
      {
        type: 'input_dummy',
        name: 'TOPROW',
      },
    ],
```

 In order to have a block with a length 0 `inputList` it would need to not have any labels or other fields on it:  
```
  // This block has no inputs.
  {
    type: 'test',
    message0: '',
    args0: [],
  },
```
![image](https://github.com/code-dot-org/code-dot-org/assets/43474485/d64ddf0d-d440-417e-a2ce-701ec7724e34)

Reply:
Should we add a comment to this effect or a no-op condition at the top with a note that we don't expect it ever to be executed?

Reply:
Sounds good. Adding and a comment and changing so we only add the button if `useModalFunctionEditor && this.inputList.length`.